grails-app/services/org/pih/warehouse/data/ProductSupplierService.groovy

```diff
@@ -208,7 +208,13 @@ class ProductSupplierService {
         if (!productSupplier) {
             productSupplier = new ProductSupplier(params)
         } else {
+            String sourceCode = productSupplier.code
             productSupplier.properties = params
+            // Do not allow to bind an empty source code, which caused the code to be re-generated for an existing product supplier (OBPIH-6288)
+            // The code could be overwritten by the binding in the 212th line, hence we'd like to bring back the existing code, if the one in params is null/empty
```

Comment:
```
in the 212th line
```
If someone changes the code above, it won't be 212th line :smile:

Reply:
Yeah let's avoid doing that, I know I did it once or twice in the past, but let's refrain from adhering to line numbers in comments.